src/config.rs

```diff
@@ -0,0 +1,74 @@
+use serde::{Deserialize, Serialize};
+use std::fs::File;
+use std::io::BufReader;
+use std::path::Path;
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct Config {
+    pub server_name: String,
+    pub sid: String,
+    pub server_description: String,
+
+    pub uplink_remote_address: String,
+    pub uplink_remote_port: u16,
+    pub uplink_password: String,
+}
+
+#[derive(Debug)]
+pub enum ConfigError {
+    InvalidSid,
+    InvalidServerName,
+    IoError(std::io::Error),
+    YamlParseError(String),
+}
+
+impl From<std::io::Error> for ConfigError {
+    fn from(e: std::io::Error) -> Self {
+        Self::IoError(e)
+    }
+}
+
+impl std::fmt::Display for ConfigError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        <Self as std::fmt::Debug>::fmt(self, f)
+    }
+}
+
+impl std::error::Error for ConfigError {}
+
+impl Config {
+    pub fn load_from_file(path: impl AsRef<Path>) -> Result<Self, ConfigError> {
+        let file = File::open(path)?;
+        let reader = BufReader::new(file);
+
+        let deserialized_config = match serde_yaml::from_reader::<BufReader<File>, Config>(reader) {
+            Ok(it) => it,
+            Err(err) => {
+                return Err(ConfigError::YamlParseError(err.to_string()));
+            }
+        };
+        return deserialized_config.validate();
```

Comment:
Consider `deserialized_config.validate()` (without the semicolon) as the last line here.